addons/mail/wizard/mail_compose_message.py

```diff
@@ -174,6 +174,35 @@ def _check_res_domain(self):
         for composer in self:
             self._parse_res_domain(composer.res_domain)
 
+    @api.depends('template_id', 'composition_mode', 'model', 'res_ids', 'parent_id', 'record_name')
+    def _compute_subject(self):
+        re_prefix = _('Re:')
+        for composer in self:
+            if composer.template_id.subject:
+                composer._set_value_from_template('subject')
+            elif not composer.template_id or not composer.subject:
+                subject = False
+                replying_subject = composer.parent_id.subject if composer.parent_id else False
```

Comment:
no need to check parent_id existence, if it's Falsy `parent_id.subject` should return False
```python
replying_subject = composer.parent_id.subject
```

(same below for `record_name`)